### PR TITLE
Matthew yung/websocket integration

### PR DESCRIPTION
**Pull request by myung03**
## Summary
Refactored code, added websockets back, and finished typing for new components

### BUGS 
- Fixed synchronous zoom bug where graphs would be synced on initial mount even when sync was false 
- Fixed Websockets returning data to all live graphs

### FEATURES 
- Re-implemented web sockets! 
- Added mock live data functionality to test data visualization (can turn on and off) 
- Refactored code in more modular packages 
    - on frontend -> folder for live data graph components, folder for db graph components, visualize/dashboard component for page (before all rendering was done in page 
    - On backend -> separated functions different files dependent on if they were db related/websocket related 
    - Finished typing new components
- Added dark mode
- Added styling changes

### Checklist
*Please change `[ ]` to `[x]` when you are ready.*
- [x] I have read and followed the code conventions detailed in [README.md](../README.md) (*This will save time for both you and the reviewer!*).
- [x] If this pull request is longer then **500** lines, I have provided *explicit* justification in the summary above explaining why I *cannot* break this up into multiple pull requests (*Small PR's are faster and less painful for everyone involved!*).

### CODE REVIEW

**Comment by Evanyl** [comment]
This dude is crazyy 🔥

File: software/tracksight/frontend/src/app/components/navbar.tsx
Comment (forbesashli) [inline software/tracksight/frontend/src/app/components/navbar.tsx:1]: love the dark mode, however it is hard my eyes (personally, I have bad eyes LOL) maybe in the future have the ability to toggle

**Review by forbesashli — APPROVED** [review]
looks great to me! really enjoy the dark mode and the changes, yeehaw

**Review by forbesashli — CHANGES_REQUESTED** [review]
Your code looks great, really like the dark mode addition. However the tests arent passing for the lower level (strange). In good practice we should investigate but it is probably an update of sorts

File: software/tracksight/backend/app/process/SignalUtil.py [hunk 1]
```diff
@@ -14,7 +14,7 @@ def __init__(self, is_mock = True):
             try:
                 self.df = pd.read_csv(MOCK_DATA_PATH)
                 self.df = self.df.set_index("date_time")
-                print(self.df.head())
+                # print(self.df.head())
```
Comment (jpatch22) [inline software/tracksight/backend/app/process/SignalUtil.py:17]: Let's just delete this, can do it in here or I can do it somewehere else

**Review by jpatch22 — APPROVED** [review]
LGTM after rebasing

File: firmware/dev/f4dev/src/cubemx/Inc/FreeRTOSConfig.h [hunk 2]
```diff
@@ -0,0 +1,177 @@
+/* USER CODE BEGIN Header */
```
Comment (jpatch22) [inline firmware/dev/f4dev/src/cubemx/Inc/FreeRTOSConfig.h:1]: Looks like some firmware stuff got added? Probably some git weirdness....

**Review by jpatch22 — CHANGES_REQUESTED** [review]

**Pull request closed without merging**